Automattic/pocket-casts-android
# Make auto download a `UserSetting`

## Description
This updates the auto download settings to each be a `UserSetting`

## Testing Instructions
1. Fresh install the app
1. Verify that the setting has the same default value that it had before
1. Change the setting to a new value
1. Force close the app
1. Verify that the setting has retained the new value
1. Verify that the app's behavior reflects the updated setting value. Note that you can force new episodes to be received from the "Developer Settings" section by tapping "Delete first episodes" and then tapping "force refresh".

## Checklist
